dask/array/core.py

```diff
@@ -3144,13 +3106,8 @@ def from_array(
             else:
                 getitem = getter_nofancy
 
-        if inline_array:
-            get_from = x
-        else:
-            get_from = original_name
-
-        dsk = getem(
-            get_from,
+        dsk = graph_from_arraylike(
```

Comment:
@rjzamora wanted to flag this as where we'll have to think through some of the array inlining logic.

Reply:
Got it.  So, the original code allows you to include the array in the graph once (with a dedicated key), or to inline it in every task.  So far, this PR is effectively inlining the array in every task by including it in the IO function (in `graph_from_arraylike`).  Is that correct?

It seems like you *could* support `inline_array=False`,  but you may need to further expand `CreateArrayDeps` to include optional args/kwargs that should be the same for all keys (and make it possible to specify these arguments in `BlockwiseCreateArray`).  I guess if you can do this, you could avoid using partial to embed the array in io funciton altogether?

Reply:
> Got it. So, the original code allows you to include the array in the graph once (with a dedicated key), or to inline it in every task. So far, this PR is effectively inlining the array in every task by including it in the IO function (in `graph_from_arraylike`). Is that correct?

> It seems like you _could_ support `inline_array=False`, but you may need to further expand `CreateArrayDeps` to include optional args/kwargs that should be the same for all keys (and make it possible to specify these arguments in `BlockwiseCreateArray`). I guess if you can do this, you could avoid using partial to embed the array in io funciton altogether?

Yes, you have this exactly right -- I'm planning to re-introduce non-inlining, but haven't tackled it yet. For the case of lazy arrays like Zarr, or small ones, this should already do what we want.